### PR TITLE
Add pod tolerations, nodeSelector and affinity in helm

### LINKED ISSUE #407: Node Affinity, Tolerations and Node selector support for helm chart
It would be nice to have more control over the deployment's behavior of agones controller by templating affinity, nodeSelector, and tolerations in our [helm chart](https://github.com/GoogleCloudPlatform/agones/blob/master/install/helm/agones/templates/controller.yaml):

variable | description | default
-- | -- | --
nodeSelector | Node labels for pod assignment | {}
affinity | Affinity settings for pod assignment | {}
tolerations | Toleration labels for pod assignment | []

See this change for example:

https://github.com/helm/charts/pull/3443/files#diff-5ae48b6a95875e88a2331216c311c529R17

### PR DESCRIPTION
**Pull request by Kuqd**
This adds pod tolerations, nodeSelector and affinity for Agones Controller and ping service as parameter variables for our helm chart.

Closes #407 

FYI: I'm also adding resource limits/requests for ping service.

### CODE REVIEW

File: install/helm/agones/templates/controller.yaml [hunk 1]
```diff
@@ -51,6 +51,18 @@ spec:
         release: {{ .Release.Name }}
         heritage: {{ .Release.Service }}
     spec:
+{{- if .Values.agones.controller.affinity }}
```
Comment (EricFortin) [inline install/helm/agones/templates/controller.yaml]:
```suggestion
    {{- if .Values.agones.controller.affinity }}
```

File: install/helm/agones/templates/ping.yaml [hunk 2]
```diff
@@ -40,10 +40,26 @@ spec:
         release: {{ .Release.Name }}
         heritage: {{ .Release.Service }}
     spec:
+{{- if .Values.agones.ping.affinity }}
```
Comment (EricFortin) [inline install/helm/agones/templates/ping.yaml]:
```suggestion
    {{- if .Values.agones.ping.affinity }}
```

File: install/helm/agones/README.md [hunk 3]
```diff
@@ -107,6 +107,9 @@ The following tables lists the configurable parameters of the Agones chart and t
 | `agones.controller.healthCheck.failureThreshold`    | Number of times before giving up (in seconds)                                                   | `3`                    |
 | `agones.controller.healthCheck.timeoutSeconds`      | Number of seconds after which the probe times out (in seconds)                                  | `1`                    |
 | `agones.controller.resources`                       | Controller resource requests/limit                                                              | `{}`                   |
+| `agones.controller.nodeSelector`                    | Controller [node labels](nodeSelector) for pod assignment                                       | `{}`                   |
```
Comment (markmandel) [inline install/helm/agones/README.md]: Can we put a :warning: on these that these are development only arguments?

File: install/helm/agones/README.md [hunk 4]
```diff
@@ -122,13 +125,20 @@ The following tables lists the configurable parameters of the Agones chart and t
 | `agones.ping.healthCheck.periodSeconds`             | ( ⚠️ release candidate feature ⚠️ ) Seconds between every liveness probe (in seconds)                  | `3`                    |
 | `agones.ping.healthCheck.failureThreshold`          | ( ⚠️ release candidate feature ⚠️ ) Number of times before giving up (in seconds)                      | `3`                    |
 | `agones.ping.healthCheck.timeoutSeconds`            | ( ⚠️ release candidate feature ⚠️ ) Number of seconds after which the probe times out (in seconds)     | `1`                    |
+| `agones.ping.resources`                             | Ping pods resource requests/limit                                                                      |`{}`                   |
```
Comment (markmandel) [inline install/helm/agones/README.md]: Ditto, :warning: on here too.

File: install/helm/agones/README.md (hunk 3 above)
Comment (markmandel) [reply install/helm/agones/README.md]:
```suggestion
| `agones.controller.nodeSelector`                    | (⚠️ ** development feature **⚠️) Controller [node labels](nodeSelector) for pod assignment                                       | `{}`                   |
```

**Pull request merged**